src/modm/ui/display/monochrome_graphic_display_vertical_impl.hpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2009-2011, 2013, 2019, Fabian Greif
+ * Copyright (c) 2011, Martin Rosekeit
+ * Copyright (c) 2012-2013, Niklas Hauser
+ * Copyright (c) 2016, Antal Szabó
+ * Copyright (c) 2021, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef MODM_MONOCHROME_GRAPHIC_DISPLAY_VERTICAL_HPP
+#error "Don't include this file directly, use 'monochrome_graphic_display_vertical.hpp' instead!"
+#endif
+
+template<int16_t Width, int16_t Height>
+void
+modm::MonochromeGraphicDisplayVertical<Width, Height>::drawHorizontalLine(glcd::Point start,
+																		  uint16_t length)
+{
+	if (start.y >= 0 and start.y < Height)
+	{
+		const int16_t y = start.y / 8;
+
+		// TODO Implement draw / clear pixels for monochrome displays
```

Comment:
What about this?

Reply:
Postponed. I'm not clear about a good solution yet. nothing is lost, it hasn't worked before.

There's lot's of stuff interacting with foreground and background color. Also there's already setPixel(), clearPixel()
writing characters intersects with these mechanisms too...
I fell like there could be a general solution that works with color and monochrome uniformly. but I want to cleanup color first #616 and afterwards spent some time on the ili9341 driver.
When i have nice color-class and a good to read ili9341 driver i want to dive into setting and clearing pixels ;)

Reply:
One possible solution: Define modm::color::Grayscale witch stores one byte with grayscale value.
For Monochrome, we only need a a boolean - storing Pixel On/Off - nut boolean will anyways consume a full byte.
So why not just using the Grayscale here too? If (value & 0x80) -> Monochrome pixel is On

All the color-types and diplays will be very compatible to each other. You will be able to use the same functions to write a colored image on a color, grayscale or monochrome display (very simplified).

```
class modm::color::Grayscale{
  uint8_t value; // Grayscale value
};
```

But yea... that's just one idea

Reply:
I think 8-bit grayscale is fine, since it's only an intermediate value anyways and it's a simple solution.
You need have an optimized image format for binary images anyways.

You can convert RGB into grayscale using this formula, however that doesn't necessarily makes sence without dithering (which definitely needs to be done beforehand).
https://github.com/salkinium/upainter/blob/master/doc/surface/README.md#8bpp-grayscale

Reply:
Cool, I check this tomorrow :)